### PR TITLE
Improve re-connect logic

### PR DESCRIPTION
**Pull request by postlund**

### CODE REVIEW

**Review by lukas-hetzenecker — CHANGES_REQUESTED** [review]

File: custom_components/remote_homeassistant/__init__.py [hunk 1]
```diff
@@ -395,8 +395,9 @@ def _async_instance_id_match(info):
         while True:
             info = await _async_instance_get_info()
 
-            # If an instance id is set, verify we are talking to correct instance
-            if not (info or _async_instance_id_match(info)):
+            # Verify we are talking to correct instance
+            if not _async_instance_id_match(info):
+                _LOGGER.error("reconnec")
```
[the remark this answers is not in the document]
Comment (postlund) [reply custom_components/remote_homeassistant/__init__.py:400]: lol, yes. That's not supposed to be there...

**Comment by postlund** [comment]
I added a few other improvements here (e.g. a version compatibility-like check). Let's see if we can get some feedback, but feel free to review.

**Comment by postlund** [comment]
The task management is pretty nuts right now btw. We should probably spend some time and try to clean that up, as it's very fragile. Lots of places where a failure can happen that breaks the re-connect flow...

**Comment by lukas-hetzenecker** [comment]
Looks great, let's merge it now - and I'll wait for some further feedback until tomorrow before publishing a new release

**Pull request merged**